src/main.c

```diff
@@ -2290,6 +2290,8 @@ H2O_NORETURN static void *run_loop(void *_thread_index)
     for (i = 0; i != conf.num_listeners; ++i) {
         if (conf.listeners[i]->quic.ctx == NULL)
             h2o_socket_read_stop(listeners[i].sock);
+        else
+            h2o_http3_close_all_connections(&listeners[i].http3.ctx.super);
```

Comment:
I'm not sure if this change is correct.

Graceful shutdown is implemented in the following way:
* step 1. Stop accepting new connections (that's what we do in this for block). This has the side effect of the newer generation of H2O accepting all new connections, when it is in fact a graceful restart.
* step 2. Send GOAWAY to all the connections, by invoking `h2o_context_request_shutdown` (see below).
* step 3. Wait for all connections to close, then exit.

What do not have code in HTTP/3 that does step 1 or step 2. Step 3 would work fine.

I think we can start from adding HTTP/3 support to step 2, as having that would be beneficial for most if not all of the deployments.

Reply:
> step 3. Wait for all connections to close, then exit.
I ran into an issue where automated tests would wait for the connection to timeout before exiting. I think the default timeout is 30s, but it felt strange that we would rely on that on shutdown. What do you think?

Reply:
I agree with you on that point. My point is that the correct fix would be to send GOAWAY thgouh `h2o_context_request_shutdown` is the correct procedure rather than abruptly closing all the connections.

I think that the issue is `h2o_context_request_shutdown` not calling the H3 request_shutdown callback, even though it calls the counterparts defined for H1 and H2.

Reply:
I'll take over this issue (#2394).